source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -217,3 +237,122 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
 
   return alarms;
 }
+
+function getSubnetIds(vpc?: ec2.IVpc, domainProps?: elasticsearch.CfnDomainProps):
+  [elasticsearch.CfnDomainProps, number?] {
+  if (vpc) {
+    // Environment-specified stacks: A ES cluster deploys in 3 AZs with 3 subnets maximum(each subnet in a different AZ).
+    // Environment-agnostic stacks: A ES cluster deploys in 2 AZs with 2 subnets maximum(each subnet in a different AZ).
+    // To successfully deploy a ES cluster, construct will need to specify exactly same subnets number as cluster AZs number
+    // To summarize: 1. Subnets number must equals to AZs number.   2. Each subnet belongs to different AZ.
+    let subnetIds = retrievePrivateSubnetIds(vpc);
+    const startPosition: number = 0;
+    let endPosition: number;
+
+    if (domainProps?.elasticsearchClusterConfig) {
+      const zoneAwareness = checkZoneAwareness(domainProps);
+      const twoAvailabilityZoneConfigFlag = checkTwoAvailabilityZoneConfig(domainProps);
+
+      if (!zoneAwareness) {
+        endPosition = 1;
+      } else if (twoAvailabilityZoneConfigFlag) {
+        endPosition = 2;
+      } else {
+        endPosition = 3;
+      }
+
+      subnetIds = selectSubnetIds(subnetIds, startPosition, endPosition);
+    } else {
+      endPosition = subnetIds.length;
+
+      if (endPosition > 3) {
+        endPosition = 3;
+      }
+
+      if (endPosition !== 2 && endPosition !== 3) {
+        throw new Error('Error - Availability Zone Count should be set to 2 or 3');
+      }
+
+      subnetIds = selectSubnetIds(subnetIds, startPosition, endPosition);
+    }
+
+    return [{
+      vpcOptions: {
+        subnetIds,
+      },
+    }, subnetIds.length];
+
+  }
+
+  return [{}];
+}
+
+function checkZoneAwareness(props: elasticsearch.CfnDomainProps): boolean | undefined {
```

Comment:
Why 

| undefined

?
You don't check the return value in the client for undefined, you just see falsy and go with false. It's set to false by default here. This should return a definitive value.